src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2881,6 +2888,176 @@ public ConstructorEdit(INamedTypeSymbol oldType)
             return semanticEdits.ToImmutable();
         }
 
+        private void TryReportFieldAttributeRudeEdits(ArrayBuilder<RudeEditDiagnostic> diagnostics, EditAndContinueCapabilities capabilities, SemanticModel? oldModel, SemanticModel newModel, Edit<SyntaxNode> edit, ArrayBuilder<SemanticEditInfo> semanticEdits, CancellationToken cancellationToken)
+        {
+            // We have to get symbols for the individual fields declared within them to check for attribute rude edits.
+            // The attributes apply to all fields in the declaration.
+            var (oldFieldSymbol, newFieldSymbol) = GetSymbolsForField(oldModel, edit.OldNode, newModel, edit.NewNode, cancellationToken);
+            if (newFieldSymbol is not null)
+            {
+                ReportAttributeEdits(oldFieldSymbol, newFieldSymbol, capabilities, diagnostics, semanticEdits, null, cancellationToken);
+            }
+        }
+
+        private void ReportAttributeEdits(ISymbol? oldSymbol, ISymbol newSymbol, EditAndContinueCapabilities capabilities, ArrayBuilder<RudeEditDiagnostic> diagnostics, ArrayBuilder<SemanticEditInfo>? semanticEdits, Func<SyntaxNode, SyntaxNode?>? syntaxMap, CancellationToken cancellationToken)
+        {
+            var needsEdit = false;
+            var attributesHaveChanged = false;
+
+            if (newSymbol is IMethodSymbol newMethod)
+            {
+                var oldMethod = oldSymbol as IMethodSymbol;
+
+                ReportAttributeRudeEdits(oldMethod?.GetReturnTypeAttributes(), newMethod.GetReturnTypeAttributes(), newMethod, capabilities, diagnostics, out attributesHaveChanged);
+                needsEdit |= attributesHaveChanged;
+
+                // For properties, we only get called for the get methods, but we want to check the property itself for attribute changes
+                if (newMethod.AssociatedSymbol is not null)
+                {
+                    ReportAttributeRudeEdits(oldMethod?.AssociatedSymbol?.GetAttributes(), newMethod.AssociatedSymbol.GetAttributes(), newMethod.AssociatedSymbol, capabilities, diagnostics, out var propertyAttributesChanged);
+                    if (propertyAttributesChanged && semanticEdits is not null)
+                    {
+                        var symbolKey = SymbolKey.Create(newMethod.AssociatedSymbol, cancellationToken);
+                        semanticEdits.Add(new SemanticEditInfo(SemanticEditKind.Update, symbolKey, syntaxMap, null, null));
+                    }
+                }
+            }
+            else if (newSymbol is INamedTypeSymbol newType)
+            {
+                var oldType = oldSymbol as INamedTypeSymbol;
+                if (newType.DelegateInvokeMethod is not null)
```

Comment:
>newType.DelegateInvokeMethod is not null [](http://example.com/codeflow?start=20&length=40)

Can this be merged into the `newSymbol is INamedTypeSymbol newType` pattern above?